# Scratch work for developing initial src/data/pipeline.py

In [1]:
import pandas as pd
from data.pipeline import build_item_store_day

In [2]:
raw_data_fp = "../data/raw"
sales_train_validation = pd.read_csv(
    f"{raw_data_fp}/sales_train_validation.csv"
)
calendar = pd.read_csv(
    f"{raw_data_fp}/calendar.csv"
)
sell_prices = pd.read_csv(
    f"{raw_data_fp}/sell_prices.csv"
)

In [3]:
item_store_day_new = build_item_store_day(
    sales_train_validation,
    calendar,
    sell_prices,
)

In [4]:
item_store_day_reference = pd.read_parquet(
    "../data/processed/item_store_day.parquet"
)

#### Validate that newly created `item_store_day` is identical to the persisted parquet one we originally created in `02_data_pipeline.ipynb`

In [5]:
# Check shape
item_store_day_new.shape == item_store_day_reference.shape

True

In [6]:
# Check column order
item_store_day_new.columns.tolist() == item_store_day_reference.columns.tolist()

True

In [7]:
# Check dtypes
item_store_day_new.dtypes.equals(item_store_day_reference.dtypes)

True

In [8]:
# Now we know the basic structure matches
# Compare the actual data
item_store_day_new.equals(item_store_day_reference)

True

In [9]:
# Everything looks good so far
# Double-check the functions output directly to make sure nothing weird happened during the comparison
item_store_day_new.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 58327370 entries, 0 to 58327369
Data columns (total 8 columns):
 #   Column         Dtype         
---  ------         -----         
 0   item_id        object        
 1   store_id       object        
 2   d              int64         
 3   date           datetime64[ns]
 4   wm_yr_wk       int64         
 5   units_sold     int64         
 6   sell_price     float64       
 7   daily_revenue  float64       
dtypes: datetime64[ns](1), float64(2), int64(3), object(2)
memory usage: 3.5+ GB


Output is as expected:
- 58,327,370 rows
- 8 columns
- all expected dtypes
- ~3.5 GB in memory
- and, importantly, it is identical to the existing validated Parquet.